### ISSUE TITLE
[Bug?] input[type=radio] value attr is not rendered if binded to empty string (FF only)

### ISSUE THREAD

**Issue by al3xnag** [emberjs/ember.js#19219]
### 🐞 Describe the Bug
`<input type="radio" value={{""}} data-foo={{""}}>` results in 
`<input type="radio" data-foo="">` in FF. The `value` attribute is missing.

In Chrome it renders as expected: `<input type="radio" data-foo="" value="">`.

### 🔬 Minimal Reproduction

https://ember-twiddle.com/5d4f5d525e3108d3a4b09142a61fe0a2?openFiles=templates.application%5C.hbs%2C

It's also reproducible on [glimmer-playground](https://glimmer-playground.netlify.app/?app=JTdCJTIyY29tcG9uZW50cyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJHbGltbWVyQXBwJTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRkdsaW1tZXJBcHAlMkZ0ZW1wbGF0ZS5oYnMlMjIlMkMlMjJzb3VyY2VUZXh0JTIyJTNBJTIyJTNDaW5wdXQlMjB0eXBlJTNEJTVDJTIycmFkaW8lNUMlMjIlMjB2YWx1ZSUzRCU3QiU3QiU1QyUyMiU1QyUyMiU3RCU3RCUyMGRhdGEtZm9vJTNEJTdCJTdCJTVDJTIyJTVDJTIyJTdEJTdEJTNFJTIyJTdEJTJDJTIyY29tcG9uZW50JTIyJTNBJTdCJTIyZmlsZU5hbWUlMjIlM0ElMjJzcmMlMkZ1aSUyRmNvbXBvbmVudHMlMkZHbGltbWVyQXBwJTJGY29tcG9uZW50LnRzJTIyJTJDJTIyc291cmNlVGV4dCUyMiUzQSUyMmltcG9ydCUyMENvbXBvbmVudCUyMGZyb20lMjAnJTQwZ2xpbW1lciUyRmNvbXBvbmVudCclM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBjbGFzcyUyMGV4dGVuZHMlMjBDb21wb25lbnQlMjAlN0IlNUNuJTdEJTIyJTdEJTdEJTVEJTJDJTIyaGVscGVycyUyMiUzQSU1QiU1RCU3RA==).

### 😕 Actual Behavior
The `value` attribute is missing in FF.

### 🤔 Expected Behavior
The `value` attribute is present and has an empty value.

### 🌍 Environment

- Ember: 3.18.1
- OS: linux
- Browser: FF 80.0.1

### ➕ Additional Context
Actually, it a problem, because 
```hbs
<input type="radio" onchange={{action (mut currentValue) value="target.value"}}>
``` 
will set `"on"` to the `currentValue` instead of an empty string, if `value` attribute is missing.

**Comment by rwjblue** [comment]
Hmm, this definitely seems like a FireFox bug here. Have you looked at their issue tracker for similar reports?

**Comment by nag5000** [comment]
No, but I can take a look.

I would like to note that natively in FF everything is ok:
`<input type="radio" value="">` renders as `<input type="radio" value="">`.

The problem appears only when `value` attr is binded to `{{""}}` expression.

**Comment by nag5000** [comment]
Probably related to #15203.

**Comment by rwjblue** [comment]
Hmm, I see. Yeah, seems odd that `value=""` works but `value={{""}}` does not. That does indeed seem like something going on here in Ember-land...

**Comment by nag5000** [comment]
I have dug into it and understood what is going on. 

*tl;dr* It seems for the fix we need to ensure [`setDynamicAttribute`](https://github.com/glimmerjs/glimmer-vm/blob/6b0db3aaa9688c2906e0f2b06baed187e9991129/packages/%40glimmer/runtime/lib/vm/element-builder.ts#L357) `-> value=""` is called after [`setStaticAttribute`](https://github.com/glimmerjs/glimmer-vm/blob/6b0db3aaa9688c2906e0f2b06baed187e9991129/packages/%40glimmer/runtime/lib/vm/element-builder.ts#L353) `-> type="radio"` in glimmer-vm.

Also I would like to ask why the [InputValueDynamicAttribute](https://github.com/glimmerjs/glimmer-vm/blob/70adc10b3b8c14c2957c06f82a53ff15b6aa8172/packages/%40glimmer/runtime/lib/vm/attributes/dynamic.ts#L170) sets and updates only a "value" property (`dom.__setProperty('value', ...)`), but not a "value" attribute. I mean, whether it is done in this way for a specific reason or not. For example, react updates both of them if `value` is binded to an expression ([codepen](https://codepen.io/gaearon/pen/VmmPgp?editors=0010)).
The fact is that it impacts what is happening here, at least in FF (maybe IE too #15203)... Maybe it's a bug in FF, but I'm not sure about it.

Take a look at what's going on step by step (in the same sequence as glimmer renders it):

| vanilla js | outerHTML Chrome | outerHTML FF |
|-|-|-|
| `i = document.createElement('input')` | `<input>` | `<input>` | 
| `i.value = ''` <br><br>(glimmer-vm `InputValueDynamicAttribute.set -> input.value = '' `) | `<input>` | `<input>` |
| `i.type = 'radio'` <br><br>(glimmer-vm `setStaticAttribute`) | `<input type="radio" value="">` 🤔 | `<input type="radio">` 🤔 <-- the reason of this bug | 

Is this a FF bug or not? I have not found anything related to this on FF bug tracker (maybe I searched badly). In the [html spec](https://html.spec.whatwg.org/multipage/input.html) I also couldn't find anything about it.

Let's keep playing with this:

| vanilla js | outerHTML Chrome | outerHTML FF |
|-|-|-|
| `i.value = 'foo'` | `<input type="radio" value="foo">` | `<input type="radio" value="foo">` |
| `i.value = ''` | `<input type="radio" value="">` | `<input type="radio" value="">` 😳 | 

And now, let's try to do `type = 'radio'` before `value = ''`:


| vanilla js | outerHTML Chrome | outerHTML FF |
|-|-|-|
| `i = document.createElement('input')` | `<input>` | `<input>` | 
| `i.type = 'radio'` <br><br>(glimmer-vm `setStaticAttribute`) | `<input type="radio">` | `<input type="radio">`| 
| `i.value = ''` <br><br>(glimmer-vm `InputValueDynamicAttribute.set -> input.value = '' `) | `<input type="radio" value="">` | `<input type="radio" value="">` 🎉🎉 |

So, in total:
  - Glimmer sets only `input.value` property and not `input[value]` attribute explicitly on newly created input element (regardless of `input[type]`. It's by design right?
  - Unlike Chrome, FF does not render empty value attribute, which was set **before** `input[type]` was changed from `text` to `radio`/`checkbox`/`...?`. Bug or not?
  - Anyway, it seems for the fix we need to ensure [`setDynamicAttribute`](https://github.com/glimmerjs/glimmer-vm/blob/6b0db3aaa9688c2906e0f2b06baed187e9991129/packages/%40glimmer/runtime/lib/vm/element-builder.ts#L357) `-> value=""` is called after [`setStaticAttribute`](https://github.com/glimmerjs/glimmer-vm/blob/6b0db3aaa9688c2906e0f2b06baed187e9991129/packages/%40glimmer/runtime/lib/vm/element-builder.ts#L353) `-> type="radio"` in glimmer-vm.

**Comment by johanrd** [comment]
this now seems reproducible in chrome 145 as well? 

<img width="740" height="425" alt="Image" src="https://github.com/user-attachments/assets/776c7e21-0eba-4589-9415-8d4a4c0f1f28" />

**Comment by NullVoxPopuli** [comment]
Looks reproducible in 6.x:

https://limber.glimdown.com/edit?c=JYWwDg9gTgLgBAYQuCA7Apq%2BAzKy4DkAAgOYA2oI6UA9AMbKQZYEDcAUKJLHAN5wwoAQzoBrdABM4AXzi58xcpWo1BI0cFQk2nFD35oZcvCEJF0IAEYqQECcGzBqO9ugAe3eBPTYhAVzJ4OjIhAGdQuAAJdDIyCAB1aDIpdxhMCQikFGZ4XnY4OCI1MUk4ADchMj90OABeOAAiBo58gQgSchr6gAoASjqAPgEAC2BQgDoKqq64AEIAIQgIMnQhVG6YUYmp6t6OAtD0GABRcBgATzq4PsGRscnK6quCFwKAHjTwELSB1oLeXibe47dDSaQALj%2BcDeIWsZAGwxgMDAoXBNBoJGAmz8lnGDBANAs1igACtQoSrNRxmSaGNQtVyQBGACcACYWQBiOnVfFULAAWgAzMzBQB2RnsiUABih70sUF%2BBSV0M0YD88AuYHQtQawnsEAacFQQioOu8tkNIJ1AKB20eoOkhoBhgIdGGaxI6AIdwmMHanTBiuVKtQao15y1Or1wANRpN2oa5tjVoaqDQ-IsYAuTt4LrdHq9PvGfo6K0D7ChbxosJiv0rlnVftQfFzzddFDE3tt40OJzO50DvbgmYuVYbSLQQarnzA33Qv2k7CAA&format=gjs


a value of `""` is present after changing the value first to something that is non-empty

**Comment by johanrd** [comment]
Possible fix: https://github.com/emberjs/ember.js/pull/21109
